### PR TITLE
chore!: replace Popper with Floating UI

### PR DESCRIPTION
**Pull request by jzempel**
**BREAKING CHANGE**: see the `migration.md` diff for details.

## Description

This PR replaces all instances of [Popper](https://popper.js.org/docs/v2/) with the new & improved [Floating UI](https://floating-ui.com/) positioning library.

## Checklist

- [ ] :ok_hand: ~design updates will be Garden Designer approved (add the designer as a reviewer)~
- [x] :globe_with_meridians: demo is up-to-date (`npm start`)
- [x] :arrow_left: renders as expected with reversed (RTL) direction
- [x] :metal: renders as expected with Bedrock CSS (`?bedrock`)
- [x] :guardsman: includes new unit tests. Maintain existing coverage (always >= 96%)
- [x] :wheelchair: tested for WCAG 2.1 AA accessibility compliance
- [x] :memo: tested in Chrome, Firefox, Safari, and Edge

### CODE REVIEW

File: packages/dropdowns.next/src/types/index.ts [hunk 1]
```diff
@@ -239,7 +222,7 @@ export interface IMenuListProps extends HTMLAttributes<HTMLUListElement> {
   /** Determines menu expansion */
   isExpanded?: boolean;
   /** Provides a list of acceptable fallback placements */
-  fallbackPlacements?: GardenPlacement[];
+  fallbackPlacements?: Exclude<GardenPlacement, 'auto'>[];
```
Comment (jzempel) [inline packages/dropdowns.next/src/types/index.ts:225]: `'auto'` does not make sense (and is not accepted by Floating UI) as a fallback placement.

File: packages/dropdowns.next/src/elements/menu/MenuList.tsx [hunk 2]
```diff
@@ -59,10 +60,10 @@ export const MenuList = forwardRef<HTMLUListElement, IMenuListProps>(
       floatingStyles: { transform }
     } = useFloating<HTMLElement>({
       elements: { reference: triggerRef?.current, floating: floatingRef?.current },
-      placement: floatingPlacement as Placement,
+      placement: floatingPlacement,
       middleware: [
         offset(theme.space.base * (hasArrow ? 2 : 1)),
-        flip({ fallbackPlacements }),
+        _placement === 'auto' ? autoPlacement() : flip({ fallbackPlacements }),
```
[the remark this answers is not in the document]
Comment (geotrev) [reply packages/dropdowns.next/src/elements/menu/MenuList.tsx:66]: This is definitely more direct. I like it.

**Comment by geotrev** [comment]
@jzempel Looks like `popperModifiers` is still exposed in the Storybook prop panel for `react-colorpickers`. Otherwise looking good!

File: packages/tooltips/src/styled/StyledTooltip.ts [hunk 3]
```diff
@@ -104,7 +104,11 @@ const sizeStyles = ({
     font-size: ${fontSize};
     overflow-wrap: ${overflowWrap};
 
-    ${hasArrow && arrowStyles(getArrowPosition(placement), { size: arrowSize, inset: arrowInset })};
+    ${hasArrow &&
+    arrowStyles(getArrowPosition(theme, placement), {
```
[the remark this answers is not in the document]
Comment (jzempel) [reply packages/tooltips/src/styled/StyledTooltip.ts:108]: yeah, prettier is overriding here.

**Comment by jzempel** [comment]
> @jzempel Looks like `popperModifiers` is still exposed in the Storybook prop panel for `react-colorpickers`. Otherwise looking good!

Hmm 🤔 I'm not seeing it https://65cfcf4a612c0b24c5c52394--zendeskgarden-react-components.netlify.app/?path=/docs/packages-colorpickers-colorswatchdialog--docs

**Comment by geotrev** [comment]
I see it in the uncontrolled/controlled demo pages. It's not in the prop sheet - which is weird, I'd expect it to be removed from the demo too. 

Screenshot 2024-02-20 at 9 30 06 AM

**Comment by geotrev** [comment]
Found it. Under `colorPickerDialog.stories.mdx` and `colorPSwatchDialog.stories.mdx`, there is an explicit `popperModifiers` set under `argTypes`.

**Review by geotrev — APPROVED** [review]

**Pull request merged**